tools/serverpod_cli/lib/src/analyzer/models/utils/quote_utils.dart

```diff
@@ -0,0 +1,43 @@
+/// Regular expression pattern that matches a string entirely enclosed
```

Comment:
Suggestion: Add examples to illustrate what each method captures.

Awesome to have all of these in a single class.
But it hard to understand what would be matched by both the regex patterns but also what would return true for the methods.
I think some examples on each case would help future developers here.